server/sources.go

```diff
@@ -31,22 +33,58 @@ type sourceLinks struct {
 
 type sourceResponse struct {
 	chronograf.Source
-	Links sourceLinks `json:"links"`
+	AuthenticationMethod string      `json:"authentication"`
+	Links                sourceLinks `json:"links"`
 }
 
-func newSourceResponse(src chronograf.Source) sourceResponse {
+type authenticationResponse struct {
+	ID                   int
+	AuthenticationMethod string
+}
+
+func sourceAuthenticationMethod(ctx context.Context, src chronograf.Source) authenticationResponse {
```

Comment:
What do you think about the REST code just calling into one of the interfaces to get this information?

I'd probably add a function to the TimeSeries interface called `Authentication`

Reply:
Any reason why the TimeSeries interfaces since the authentication method is tied to the source?

Reply:
Source is a struct and TimeSeries is an interface that provides functions around all the difference time series databases.

However, you could create another interface.  Smaller interfaces are good. 
Perhaps something like :

```
type AuthenticationType int

const (
  BasicAuth AuthenticationType = iota
  BearerAuth,
  LDAPAuth,
  TokenAuth, // this is for the next API in platform repo
)

type Authenticator interface {
   AuthType(context.Context) (AuthenicationType, error)
}
```

... then the enterprise and influx could implement that function.